### PR TITLE
Add required `mapping` keyword to entitlement mapping usages

### PR DESCRIPTION
**Pull request by dsainati1**
Implementation of https://github.com/onflow/flips/pull/210
______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

**Review by turbolent — COMMENTED** [review]
Nice!

I assumed the changes would be mostly syntactical, but there are also some changes to the checking unrelated (?) to the new `mapping` keyword?

File: runtime/sema/checker.go [hunk 1]
```diff
@@ -1256,19 +1261,21 @@ func (checker *Checker) functionType(
...
 	returnTypeAnnotation *ast.TypeAnnotation,
 ) *FunctionType {
+
+	oldMappedAccess := checker.entitlementMappingInScope
+	if mapAccess, isMapAccess := access.(*EntitlementMapAccess); isMapAccess {
+		checker.entitlementMappingInScope = mapAccess.Type
+	} else {
+		checker.entitlementMappingInScope = nil
+	}
+	defer func() { checker.entitlementMappingInScope = oldMappedAccess }()
```
[the remark this answers is not in the document]
Comment (dsainati1) [reply runtime/sema/checker.go:1271]: Oh that's odd, these changes are from https://github.com/onflow/cadence/pull/2877, they seem to have found their way in here by accident, maybe when I saved a file from that branch on this one.

**Comment by dsainati1** [comment]
Yeah, the changes should be entirely syntactic, with the exception of a few checker-related changes in converting ast-level access to semantic access, since we no longer need to disambiguate between a singleton entitlement and a map.

File: runtime/tests/checker/entitlements_test.go [hunk 2]
```diff
@@ -1535,52 +1509,19 @@ func TestCheckBasicEntitlementMappingAccess(t *testing.T) {
...
-	t.Run("multiple mappings disjunction", func(t *testing.T) {
-		t.Parallel()
-
-		_, err := ParseAndCheck(t, `
-            entitlement mapping M {}
-            entitlement mapping N {}
+            entitlement M
             resource interface R {
-                access(M | N) let foo: String
+                access(mapping M) let foo: String
```
Comment (SupunS) [inline runtime/tests/checker/entitlements_test.go:1518]: Is there a test for the opposite: Using an entitlement mapping, without the `mapping` keyword?

**Review by SupunS — APPROVED** [review]

File: runtime/tests/checker/entitlements_test.go (hunk 2 above)
Comment (dsainati1) [reply runtime/tests/checker/entitlements_test.go:1518]: Added

File: runtime/ast/type.go [hunk 3]
```diff
@@ -577,25 +573,35 @@ func (t *ReferenceType) EndPosition(memoryGauge common.MemoryGauge) Position {
...
+		case *MappedAccess:
+			doc = append(doc,
+				referenceTypeMappingKeywordDoc,
+				authorization.EntitlementMap.Doc(),
+			)
+		default:
+			panic(errors.NewUnreachableError())
 		}
+		doc = append(doc, prettier.Text(")"))
 		doc = append(doc, prettier.Space)
```
Comment (turbolent) [inline runtime/ast/type.go:605]:
```suggestion
		doc = append(doc, 
			prettier.Text(")", 
			prettier.Space,
		)
```

File: runtime/parser/type.go [hunk 4]
```diff
@@ -921,11 +921,30 @@ func defineIdentifierTypes() {
...
+					keywordPos := p.current.StartPos
+					// Skip the keyword
+					p.nextSemanticToken()
+
+					entitlementMapName, err := parseNominalType(p, lowestBindingPower)
+					if err != nil {
+						return nil, err
+					}
+					authorization = ast.NewMappedAccess(entitlementMapName, keywordPos)
+					p.skipSpaceAndComments()
```
Comment (turbolent) [inline runtime/parser/type.go:938]: Is this needed?

File: runtime/sema/errors.go [hunk 5]
```diff
@@ -4279,6 +4280,22 @@ func (e *InvalidNonEntitlementAccessError) Error() string {
 	return "only entitlements may be used in access modifiers"
 }
 
+// MappingAccessMissingKeywordError
+type MappingAccessMissingKeywordError struct {
```
Comment (turbolent) [inline runtime/sema/errors.go:4284]: Maybe add a suggested fix

File: runtime/sema/errors.go [hunk 6]
```diff
@@ -4190,27 +4190,28 @@ func (e *InvalidEntitlementAccessError) EndPosition(common.MemoryGauge) ast.Posi
 	return e.Pos
 }
 
-// InvalidMultipleMappedEntitlementError
-type InvalidMultipleMappedEntitlementError struct {
-	Pos ast.Position
+// InvalidEntitlementMappingTypeError
+type InvalidEntitlementMappingTypeError struct {
```
Comment (turbolent) [inline runtime/sema/errors.go:4194]: Maybe add a suggested fix

**Review by turbolent — APPROVED** [review]
Nice! Just a couple minor suggestions and questions

File: runtime/parser/type.go (hunk 4 above)
Comment (dsainati1) [reply runtime/parser/type.go:938]: Yes, nominal type parsing does not skip space/comments at the end

**Pull request merged**